backend/src/main/java/saci/configuration/ControllerAdvice.java

```diff
@@ -0,0 +1,43 @@
+package saci.configuration;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import saci.domain.service.exceptions.AlreadyExistsException;
+import saci.domain.service.exceptions.CoefficientOverlapException;
+import saci.domain.service.exceptions.NotFoundException;
+
+@Slf4j
+@RestControllerAdvice
+public class ControllerAdvice {
+
+    @ExceptionHandler(NotFoundException.class)
+    public ResponseEntity<String> handleNotFoundException(NotFoundException ex) {
+        String errorMessage = "Resource not found: " + ex.getMessage();
+        log.error(errorMessage, ex);
```

Comment:
```suggestion
        log.error(errorMessage);
```
Just the errorMessage is enough for all exceptions